### PR TITLE
refactor: migrate to ruff

### PR DESCRIPTION
**Pull request by aciba90**
Refactor black, isort, flake8 to use ruff.
Apply ruff format.

Fixes: GH-373

### CODE REVIEW

**Comment by aciba90** [comment]
> what is the benefit of using ruff vs black and isort?

Generally the advantage of ruff is speed. In the case of black and isort there is not a huge difference, but having a single tool reduces complexity and dependencies.

File: pyproject.toml [hunk 1]
```diff
@@ -31,3 +27,34 @@ module = [
   "paramiko.*",
 ]
 ignore_missing_imports = true
+
```
Comment (TheRealFalcon) [inline pyproject.toml:30]: We should remove the `[tool.black]` section too, correct?

File: pyproject.toml [hunk 2]
```diff
@@ -31,3 +27,34 @@ module = [
...
+  "I",  # isort
+  "PL",  # pylint
+]
+
+# Commented rules were not available in ruff. Enable them as they are added to ruff
+# See: https://github.com/astral-sh/ruff/issues/970
+ignore = [
+  "N815",  # Invalid name
+  "SIM300",  # misplaced-comparison-constant
+  # "PLC0209",  # Formatting a regular string which could be a f-string (consider-using-f-string)
```
Comment (TheRealFalcon) [inline pyproject.toml:49]: Were the commented lines carried over from flake8? Can we remove them now?

File: .pylintrc [hunk 3]
```diff
@@ -1,20 +0,0 @@
-[MASTER]
```
Comment (TheRealFalcon) [inline .pylintrc:1]: Pylint still does a fair number of things that ruff cannot (and probably never will). I don't think we should be removing pylint here.

**Review by TheRealFalcon — CHANGES_REQUESTED** [review]
Thanks for this! See my inline comments.

File: pyproject.toml (hunk 1 above)
Comment (aciba90) [reply pyproject.toml:26]: Removed.

File: pyproject.toml (hunk 2 above)
Comment (aciba90) [reply pyproject.toml]: Those were not yet available pylint rules, see the header comment. This is not needed anymore as we reverted to keep pylint.

File: .pylintrc (hunk 3 above)
Comment (aciba90) [reply .pylintrc]: Reverted.

File: tox.ini [hunk 4]
```diff
@@ -106,13 +97,8 @@ deps = {[testenv:integration-tests-ci]deps}
...
-[flake8]
-# E203: whitespace before ':' ... This goes against pep8 and black formatting
-# W503: line break before binary operator
-ignore = E203, W503
-
 [pytest]
 testpaths = tests/unit_tests
 markers =
-    ci: run test on as part of continous integration
+    ci: run test on as part of continouos integration
```
Comment (TheRealFalcon) [inline tox.ini:103]: Nit:
```suggestion
    ci: run test on as part of continuous integration
```

**Review by TheRealFalcon — APPROVED** [review]
One nit inline, but otherwise LGTM!

File: tox.ini (hunk 4 above)
Comment (aciba90) [reply tox.ini]: Thanks, fixed.

**Pull request merged**